sim_instr/src/sim_instr_node.cpp

```diff
@@ -41,8 +44,12 @@ int main(int argc, char** argv) {
     ros::NodeHandle nh("~");
     ros::Publisher pub = nh.advertise<geometry_msgs::Twist>("cmd_vel", 1000);
 
+	while (pub.getNumSubscribers() == 0) {
+		ros::spinOnce();	
+	}
+
     std::string filename;
-    if (!nh.getParam("filename", filename)) {
+    if (!nh.getParam("param", filename)) {
```

Comment:
shouldn't this be filename? why would you provide the filename like `rosrun sim_instr sim_instr_node _param:=somefile.txt`?